### PR TITLE
Minor flyte role updates

### PR DESCRIPTION
**Pull request by noahjax**
### Jira
https://dominodatalab.atlassian.net/browse/DOM-54483

### Description
Initial roles had a few minor issues that are cleaned up here

### CODE REVIEW

File: modules/flyte/iam.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ resource "aws_iam_role" "flyte_controlplane" {
           StringEquals : {
             "${trimprefix(local.oidc_provider_url, "https://")}:aud" : "sts.amazonaws.com",
             "${trimprefix(local.oidc_provider_url, "https://")}:sub" : [
-              "system:serviceaccount:${var.platform_namespace}:${var.serviceaccount_names.flyteadmin}",
+              "system:serviceaccount:${var.platform_namespace}:${var.serviceaccount_names.datacatalog}",
```
Comment (noahjax) [inline modules/flyte/iam.tf:16]: Leaving datacatalog with controlplane role for now until we confirm that it needs data bucket access

File: modules/flyte/iam.tf [hunk 2]
```diff
@@ -60,15 +60,29 @@ resource "aws_iam_role" "flyte_dataplane" {
           Federated = local.oidc_provider_arn
         }
         Condition : {
-          StringEquals : {
+          StringLike : {
```
Comment (noahjax) [inline modules/flyte/iam.tf:63]: Oversight on my part in the initial PR, `StringEquals` doesn't work with the wildcard

File: modules/flyte/iam.tf [hunk 3]
```diff
@@ -60,15 +60,29 @@ resource "aws_iam_role" "flyte_dataplane" {
...
             "${trimprefix(local.oidc_provider_url, "https://")}:aud" : "sts.amazonaws.com",
             "${trimprefix(local.oidc_provider_url, "https://")}:sub" : [
-              "system:serviceaccount:${var.platform_namespace}:${var.serviceaccount_names.datacatalog}",
               "system:serviceaccount:${var.compute_namespace}:*"
```
[the remark this answers is not in the document]
Comment (noahjax) [reply modules/flyte/iam.tf:66]: Good point, no need to worry about supporting flyte native tasks for now

**Review by ddl-rliu — APPROVED** [review]

File: modules/flyte/iam.tf (hunk 1 above)
Comment (noahjax) [reply modules/flyte/iam.tf:16]: I think you're right here; input file uploads are just wonky ([issue](https://github.com/flyteorg/flyte/issues/5051))

**Review by Secretions — APPROVED** [review]

**Pull request merged**